src/test/java/hudson/plugins/active_directory/docker/TheFlintstonesTest.java

```diff
@@ -98,7 +98,11 @@ public void overrideDNS() throws Exception {
         // see hudson.plugins.active_directory.ActiveDirectoryDomain.createDNSLookupContext()
         // getHost returns a hostname not IPaddress...
         // use our DNS to resolve that to an IP address.
-        String DNS_URLs = "dns://"+InetAddress.getByName(docker.getHost()).getHostAddress()+":"+docker.getDNSPort();
+        String hostAddress = InetAddress.getByName(docker.getHost()).getHostAddress();
+        if (Boolean.getBoolean("java.net.preferIPv6Addresses")) {
```

Comment:
I think this would break on IPv6 only hosts? 
as you are getting an InetAddress, you can already know if you have a v4 or v6 address and react accordingly and do not use the system property.

e.g.
```java
InetAddress inAddr = InetAddress.getByName(docker.getHost());
if (inAddr instanceof Inet6Address) {
 //...
 } else {
  /....
}
```
now if you want to do something based on a preference, you should be asking for **all** addresses (`getAllByName()`), and looking for the one according to the preference - but there may not be one...  But YAGNI


or just stop building URLs by hand... and embrace the APIs
`new URI("dns", null, docker.getHost()).getHostAddress(), docker.getDNSPort(), null, null, null).toASCIIString();`

Reply:
Addressed in https://github.com/jenkinsci/active-directory-plugin/pull/183/commits/7690fd7ebd7056af5971ada488e892b781d5db3a